pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -533,15 +534,36 @@ class ModelValidatorImpl implements ModelValidator {
 
         if (agent.args instanceof ModelASTSingleArgument) {
             ModelASTSingleArgument singleArg = (ModelASTSingleArgument) agent.args
+            Map<String,DescribableModel> zeroArgModels = DeclarativeAgentDescriptor.zeroArgModels()
+            if (!zeroArgModels.containsKey(singleArg.value.getValue())) {
+                errorCollector.error(agent.args, "Invalid argument for agent - '${singleArg.value.toGroovy()}' - must be map of config options or bare ${zeroArgModels.keySet().sort()}.")
+                valid = false
+            }
+        } else if (agent.args instanceof ModelASTNamedArgumentList) {
+            ModelASTNamedArgumentList namedArgs = (ModelASTNamedArgumentList)agent.args
+            List<String> argKeys = namedArgs.arguments.collect { k, v ->
+                k.key
+            }
```

Comment:
Could just collapse the two cases here, passing `[:]` in case of `ModelASTSingleArgument`.